MoltenVK/MoltenVK/GPUObjects/MVKRenderPass.h

```diff
@@ -130,12 +130,13 @@ class MVKRenderSubpass : public MVKBaseObject {
 							const MVKArrayRef<MVKImageView*>& attachments,
 							bool storeOverride = false);
 
-	/** Constructs an instance for the specified parent renderpass. */
+	/** Resolves any resolve attachments that cannot be handled by native Metal subpass resolve behavior. */
+	void resolveUnresolvableAttachments(MVKCommandEncoder* cmdEncoder, const MVKArrayRef<MVKImageView*>& attachments);
```

Comment:
Since an `MVKArrayRef<T>` is just a pointer and a size, passing it around by reference doesn't buy much. It might actually hurt performance, since cache lines might be evicted to make room for the `MVKArrayRef` in memory. For this reason, LLVM passes its `ArrayRef`s by value. I think we should consider passing this by value instead of by reference in the future.

Reply:
Thanks. Interesting technical detail observation.

No time like the present! I've changed all existing calls passing `MVKArrayRef` to use pass-by-value.